### PR TITLE
Add `.isFromCache` property to the stream API and rename the Promise API property from `.fromCache` to `.isFromCache`

### LINKED ISSUE #747: Stream `fromCache` property
I need to `pipe()` the downloaded response for further processing (unzipping), but only when the response is not from cache. However, when using `got.stream()`, I can't figure out how to check where the response comes from - origin or cache.

```js
...
const response = got.stream(originUrl, { cache }); 
const fromCache = ...; // How to set this
if (fromCache) {
  response.pipe(unzip). ...
}
```
I've also tried
```js
const response = await got(originUrl, { cache });

if (response.fromCache) { // Here cache check is ok
  response.pipe(unzip). ...
}
```
but in this case response stream is already consumed.

Thanks

#### Checklist

- [x] I have read the documentation.

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes #747 

TODO: docs

### CODE REVIEW

File: source/as-stream.ts [hunk 1]
```diff
@@ -37,7 +42,14 @@ export default function asStream(options: MergedOptions) {
...
 
-		isFinished = true;
+		{
+			const read = proxy._read.bind(proxy);
+			proxy._read = (...args) => {
+				isFinished = true;
+
+				return read(...args);
+			};
+		}
```
Comment (szmarczak) [inline source/as-stream.ts]: It's a pity there is no `read` event...

Comment (szmarczak) [reply source/as-stream.ts]: After the first chunk is sent further `.pipe(...)` calls should throw.

Currently it throws if the `response` event was emitted.

**Comment by sindresorhus** [comment]
I would name it `.isFromCache`.

Is there any point in making it `undefined` before the `request` event? Can't we just make it `false` then?

**Comment by szmarczak** [comment]
> I would name it .isFromCache.

Sure. Should it be changed for Promises too?

> Is there any point in making it undefined before the request event? Can't we just make it false then?

It's undefined because we don't know is it true or false. I don't wanna make users mistaken.

**Comment by sindresorhus** [comment]
Yes, let’s change the Promise one too (forgot we had it there).

**Comment by sindresorhus** [comment]
👌

**Pull request merged**